### PR TITLE
Finalize keypoint detection

### PR DESCRIPTION
**Pull request by SolomonLake**
# Description

Updates keypoint detection to match with what is being sent from dataset/weights routes. Also adds validation for num_classes.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

Tested running local inference server for one keypoints detection model and one OD model.

## Any specific deployment considerations

None.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/roboflow_api.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "object-detection": "yolov5v2s",
     "instance-segmentation": "yolact",
     "classification": "vit",
+    "keypoint-detection": "yolov8n",
```
Comment (SolomonLake) [inline inference/core/roboflow_api.py:38]: most changes are `keypoints-detection` -> `keypoint-detection`, but this was a necessary addition to make sure model manager accepts type.

Comment (PawelPeczek-Roboflow) [reply inference/core/roboflow_api.py:38]: 👍

**Review by PawelPeczek-Roboflow — APPROVED** [review]

File: inference/core/models/keypoints_detection_base.py [hunk 2]
```diff
@@ -188,3 +189,27 @@ def _model_keypoints_to_response(
...
+        return max_keypoints
+
+    def validate_model_classes(self) -> None:
+        num_keypoints = self.superset_keypoints_count()
+        keypoints_shape = num_keypoints * 3
+        output_shape = self.get_model_output_shape()
+        num_classes = get_num_classes_from_model_prediction_shape(
+            len_prediction=output_shape[2], keypoints_shape=keypoints_shape
+        )
+        try:
```
Comment (PawelPeczek-Roboflow) [inline inference/core/models/keypoints_detection_base.py:210]:
```python
if num_classes != self.num_classes:
    raise ValueError()
```

**Review by PawelPeczek-Roboflow — APPROVED** [review]

File: tests/inference/unit_tests/core/models/utils/test_keypoints.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            14: "right_knee",
+            15: "left_ankle",
+            16: "right_ankle",
+        }
+    }
+    keypoints_count = superset_keypoints_count(keypoints_metadata)
+    assert keypoints_count == 17
+
+
+def test_superset_keypoints_count_with_two_classes():
```
Comment (PawelPeczek-Roboflow) [inline tests/inference/unit_tests/core/models/utils/test_keypoints.py:32]: Suggestion:
Return type: `-> None:`

given-when-then
```python
# given
keypoints_metadata = ...

# when
keypoints_count = superset_keypoints_count(keypoints_metadata)

# then
assert keypoints_count == 13, "Keypoints of object 1 form a superset of all classes"
```

Would be also good to cover the behaviour when classes do not overlap

**Review by PawelPeczek-Roboflow — APPROVED** [review]
:+1

File: tests/inference/unit_tests/core/models/utils/test_keypoints.py (hunk 3 above)
Comment (SolomonLake) [reply tests/inference/unit_tests/core/models/utils/test_keypoints.py]: Actually, maybe I need another review? merging is blocked

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**